Development-Guide/Compile-and-Flash-STMBoard-Guide/README.adoc

```diff
@@ -0,0 +1,75 @@
+= Compiling and Flashing STM Board
+
+NOTE: This has only been tested on Linux Mint, but is expected to work on other Debian based distros as well.
+
+== Setup
+
+. Go to UCSolarCarTeam repo > Recruit Resources, and clone.
+
+. Navigate to STM32F4Discovery-Resources and run the auto installer `sudo ./install.sh`. 
+This will should download STLink, gcc4mbed, and CubeMX2Makefile and add them all to the path.
+
+. Check to make sure all of them were added by using a preferred text editor.
+For example, `nano ~/.profile`.
+If any of them are missing add them to the path.
+
+.. Add them by copying  `export PATH=$PATH:/opt/CubeMX2Makefile`, `export PATH=$PATH:/opt/stlink`, or `export PATH=$PATH:/opt/gcc4mbed`  in `.profile`.
+
+.. Save, exit, and source `source ~/.profile`.
+
+. http://www.st.com/en/development-tools/stm32cubemx.html[Go to the download page of the official ST website by clicking this link^].
+
+. Scroll to the bottom of the page to the *GET SOFTWARE* section and click *Get Software*.
+
+. Follow the instructions and download CubeMX.
+
+. Extract files from `.zip` file using a preferred method.
+For example `unzip file.zip -d Destination_folder`.
+
+. Navigate to the directory the files were extracted to.
+
+. Run *SetupSTM32CubeMX-VERSION.linux* `./SetupSTM32CubeMX-{var}.linux` where  {var} is the version of CubeMx that is being installed.
+
+. Follow the instructions in the Installation Wizard.
+
+
+== Generating Code with CubeMX
+
+. Open CubeMX. The default path is `/home/STM32CubeMX`.
+
+. Select New Project.
+
+. Click on the Board Selector Tab and scroll to STM32F4DISCOVERY then click OK.
+
+. Go to Project > Project Settings.
+
+. Name the project, set the project location, and change the Toolchain/IDE to SW4STM32. 
+*If this isn't done, CubeMX2Makefile won't work!* 
+CubeMX2Makefile looks for an SW4STM32 project which correlates to CubeMX.
+If the Toolchain/IDE is not set to SW4STM32, it will be in the format of a different IDE, not CubeMX.
+For example, EWARM is for the IAR Embedded Workbench IDE and CubeMX2Makefile will not know what to do with that format.
+
+. Make note of the Toolchain Folder Location, which will also be the project folder location.
+
+. Click OK and download any dependencies that are needed, then click OK again.
+
+. Do configurations and go to Project > Generate Code, then save project.
+
+
+== Generating a Makefile and Compiling
+. Create the makefile  `CubeMX2Makefile /path/to/project/folder`.
```

Comment:
When you're going to give commands you can't just put them right beside some text like it's part of the sentence. You either need to separate it from the sentence or integrate it. One way to separate it is putting it on a new line. Integrating it would look like 

"Create the Makefile using `CubeMX2Makefile /path/to/project/folder`."